sql/2021/ecommerce/core_web_vitals_passingmetrics_byvendor_bydevice.sql

```diff
@@ -0,0 +1,64 @@
+#standardSQL
+# CrUX Core Web Vitals performance of Ecommerce vendors by device
+CREATE TEMP FUNCTION IS_GOOD (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good / (good + needs_improvement + poor) >= 0.75
+);
+
+CREATE TEMP FUNCTION IS_NON_ZERO (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good + needs_improvement + poor > 0
+);
+
+
+SELECT
+  client,
+  ecomm,
+  COUNT(DISTINCT origin) AS origins,
+  # Origins with good LCP divided by origins with any LCP.
+  SAFE_DIVIDE(
+    COUNT(DISTINCT IF(IS_GOOD(fast_lcp, avg_lcp, slow_lcp), origin, NULL)),
+    COUNT(DISTINCT IF(IS_NON_ZERO(fast_lcp, avg_lcp, slow_lcp), origin, NULL))) AS pct_good_lcp,
+
+  # Origins with good FID divided by origins with any FID.
+  SAFE_DIVIDE(
+    COUNT(DISTINCT IF(IS_GOOD(fast_fid, avg_fid, slow_fid), origin, NULL)),
+    COUNT(DISTINCT IF(IS_NON_ZERO(fast_fid, avg_fid, slow_fid), origin, NULL))) AS pct_good_fid,
+
+  # Origins with good CLS divided by origins with any CLS.
+  SAFE_DIVIDE(
+    COUNT(DISTINCT IF(IS_GOOD(small_cls, medium_cls, large_cls), origin, NULL)),
+    COUNT(DISTINCT IF(IS_NON_ZERO(small_cls, medium_cls, large_cls), origin, NULL))) AS pct_good_cls,
+
+  # Origins with good LCP, FID, and CLS dividied by origins with any LCP, FID, and CLS.
```

Comment:
Since many sites have no FID data, in our CWV evaluations we should consider it optional.

See the CWV Tech Report queries for an example of calculating the [numerator](https://github.com/HTTPArchive/cwv-tech-report/blob/50705bab1e712218c679e86ffac9995fb650452e/sql/monthly.sql#L56-L58) and [denominator/pct](https://github.com/HTTPArchive/cwv-tech-report/blob/50705bab1e712218c679e86ffac9995fb650452e/sql/monthly.sql#L121)